RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitWorkerSimple.cc

```diff
@@ -9,111 +9,133 @@
 #include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
 
 HGCalRecHitWorkerSimple::HGCalRecHitWorkerSimple(const edm::ParameterSet&ps) :
-  HGCalRecHitWorkerBaseClass(ps) {
-  rechitMaker_.reset( new HGCalRecHitSimpleAlgo() );
-  constexpr float keV2GeV = 1e-6;
-  // HGCee constants 
-  HGCEE_keV2DIGI_   =  ps.getParameter<double>("HGCEE_keV2DIGI");
-  HGCEE_fCPerMIP_   =  ps.getParameter<std::vector<double> >("HGCEE_fCPerMIP");
-  HGCEE_isSiFE_     =  ps.getParameter<bool>("HGCEE_isSiFE");
-  hgceeUncalib2GeV_ = keV2GeV/HGCEE_keV2DIGI_;
-  
-  // HGChef constants
-  HGCHEF_keV2DIGI_   =  ps.getParameter<double>("HGCHEF_keV2DIGI");
-  HGCHEF_fCPerMIP_   =  ps.getParameter<std::vector<double> >("HGCHEF_fCPerMIP");
-  HGCHEF_isSiFE_     =  ps.getParameter<bool>("HGCHEF_isSiFE");
-  hgchefUncalib2GeV_ = keV2GeV/HGCHEF_keV2DIGI_;
-  
-  // HGCheb constants
-  HGCHEB_keV2DIGI_   =  ps.getParameter<double>("HGCHEB_keV2DIGI");
-  HGCHEB_isSiFE_     =  ps.getParameter<bool>("HGCHEB_isSiFE");
-  hgchebUncalib2GeV_ = keV2GeV/HGCHEB_keV2DIGI_;
-
-  // layer weights (from Valeri/Arabella)
-  std::vector<float> weights;
-  const auto& dweights = ps.getParameter<std::vector<double> >("layerWeights");
-  for( auto weight : dweights ) {
-    weights.push_back(weight);
-  }
-  rechitMaker_->setLayerWeights(weights);
-
-  // residual correction for cell thickness
-  const auto& rcorr = ps.getParameter<std::vector<double> >("thicknessCorrection");
-  rcorr_.clear();
-  rcorr_.push_back(1.f);
-  for( auto corr : rcorr ) {
-    rcorr_.push_back(1.0/corr);
-  }
-  
+        HGCalRecHitWorkerBaseClass(ps)
+{
+    rechitMaker_.reset(new HGCalRecHitSimpleAlgo());
+    tools_.reset(new hgcal::RecHitTools());
+    constexpr float keV2GeV = 1e-6;
+    // HGCee constants
+    HGCEE_keV2DIGI_ = ps.getParameter<double>("HGCEE_keV2DIGI");
+    HGCEE_fCPerMIP_ = ps.getParameter < std::vector<double> > ("HGCEE_fCPerMIP");
+    HGCEE_isSiFE_ = ps.getParameter<bool>("HGCEE_isSiFE");
+    hgceeUncalib2GeV_ = keV2GeV / HGCEE_keV2DIGI_;
+
+    // HGChef constants
+    HGCHEF_keV2DIGI_ = ps.getParameter<double>("HGCHEF_keV2DIGI");
+    HGCHEF_fCPerMIP_ = ps.getParameter < std::vector<double> > ("HGCHEF_fCPerMIP");
+    HGCHEF_isSiFE_ = ps.getParameter<bool>("HGCHEF_isSiFE");
+    hgchefUncalib2GeV_ = keV2GeV / HGCHEF_keV2DIGI_;
+
+    // HGCheb constants
+    HGCHEB_keV2DIGI_ = ps.getParameter<double>("HGCHEB_keV2DIGI");
+    HGCHEB_isSiFE_ = ps.getParameter<bool>("HGCHEB_isSiFE");
+    hgchebUncalib2GeV_ = keV2GeV / HGCHEB_keV2DIGI_;
+
+    // layer weights (from Valeri/Arabella)
+    const auto& dweights = ps.getParameter < std::vector<double> > ("layerWeights");
```

Comment:
no need for a loop and `push_back`, just do:
```
weights_ = ps.getParameter < std::vector<double> > ("layerWeights");
```

In fact, all the member variables being initialized from parameter set values can just go into the member initializer list of the constructor.